src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -84,24 +86,54 @@ def check_metadata(self):
         validate(metadata_for_validation, schema)
         self.check_extracted_metadata(metadata)
 
-    def check_run_conversion(self, nwbfile_path: str):
+    def check_no_metadata_mutation(self):
```

Comment:
Other thing being that we don't need to actually write a file to disk to check this behavior - it's enough to just call the `.add_to_nwbfile` protocol and check the `metadata` argument in and out

Reply:
Thanks. This is indeed better.